### ISSUE TITLE
Enums as generics fail to be imported in declaration output

### ISSUE THREAD

**Issue by Yuudaari** [Microsoft/TypeScript#27962]
**TypeScript Version:**  3.2.0-dev.20181017

**Search Terms:** enum declaration generic type argument import

When a field/variable is set via its value, and the value's type contains an enum as a type argument, the enum is not imported in the declaration for that file. This is kind of confusing to explain, so I'll make it more clear with the following example:

### Code

**TestEnumAndClass.ts**
This file creates the enum and a few things which use the enum as a generic.

```ts
export enum TestEnum {
	Foo,
	Bar,
	Baz
}

export default class GenericClass<T extends TestEnum = TestEnum> {
	public constructor(test: T) {}
}

export type GenericType<T extends TestEnum = TestEnum> = Array<T> | IterableIterator<T> | Set<T> | Map<T, T>;

export const genericInstance = new GenericClass(TestEnum.Foo);
export let genericArray: GenericType<TestEnum.Foo>;
```

**test.ts**
This file imports some of the things using the enum as a generic, but not the enum.

```ts
import GenericClass, { genericInstance } from "./TestEnumAndClass";

export const tests = {
	foo: genericInstance,
	bar: {} as any as GenericClass,
	baz: genericArray,
	fizz: {} as any as GenericType
};

export const buzz = genericInstance;
```

### Expected behavior:
The enum should be imported, since it's required to correctly type the variables. If not imported this way, imported via an inline `import()` call.
```ts
// test.d.ts
import GenericClass, { GenericType, TestEnum } from "./TestEnumAndClass";
export declare const tests: {
    foo: GenericClass<TestEnum.Foo>;
    bar: GenericClass<TestEnum>;
    baz: GenericType<TestEnum.Foo>;
    fizz: GenericType<TestEnum>;
};
export declare const buzz: GenericClass<TestEnum.Foo>;
```

### Actual behavior:
The enum is not imported at all. This causes the declaration file to be invalid.
```ts
// test.d.ts
import GenericClass, { GenericType } from "./TestEnumAndClass";
export declare const tests: {
    foo: GenericClass<TestEnum.Foo>;
    bar: GenericClass<TestEnum>;
    baz: GenericType<TestEnum.Foo>;
    fizz: GenericType<TestEnum>;
};
export declare const buzz: GenericClass<TestEnum.Foo>;
```

**Comment by mattmccutchen** [comment]
Another duplicate of #26969?

**Comment by djskinner** [comment]
It seems 3.1.5 did not fix this so looks to be a slightly different issue. 

I'm not sure if my reproduction is exactly the same but I can confirm I'm still having an issue with declaration files missing imports on enums when generics are involved on 3.1.5.

Happy to add another repro but I suspect the original will be sufficient.

**Comment by weswigham** [comment]
The fix for this should be in the 3.2 train (ie, on @next), not 3.1.5. 3.1.5 only had a choice few bugfixes.

**Comment by rix0rrr** [comment]
I can reproduce this without generics as well, on `@next`:

```ts
// provider.ts
export enum Enum {
    Value1,
    Value2,
}
```

```ts
// consumer.ts
import provider = require('./provider');

export const value = provider.Enum.Value1;
```

```
$ node_modules/.bin/tsc --version
Version 3.2.0-dev.20181107

$ node_modules/.bin/tsc --declaration *.ts
Found 0 errors.

$ cat consumer.d.ts 
export declare const value = provider.Enum.Value1;
```

Missing `import`.

Adding an explicit type declaration fixes it:

```ts
// consumer.ts
import provider = require('./provider');

export const value: provider.Enum = provider.Enum.Value1;
```

Leads to

```ts
// consumer.d.ts
import provider = require('./provider');
export declare const value: provider.Enum;
```
